superset-frontend/src/SqlLab/components/QuerySearch.tsx

```diff
@@ -0,0 +1,283 @@
+import { debounce } from 'lodash';
+import Loading from '../../components/Loading';
+import QueryTable from './QueryTable';
+import {
+  now,
+  epochTimeXHoursAgo,
+  epochTimeXDaysAgo,
+  epochTimeXYearsAgo,
+} from '../../modules/dates';
+import { STATUS_OPTIONS, TIME_OPTIONS } from '../constants';
+import AsyncSelect from '../../components/AsyncSelect';
+
+interface propTypes {
+  actions: {
+    addDangerToast: (msg: string) => void;
+    setDatabases: (data: Record<string, any>) => Record<string, any>;
+  };
+  displayLimit: number;
+}
+
+interface userMutatorProps {
+  value: number;
+  text: string;
+}
+
+const TableWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex: 1;
+  height: 100%;
+`;
+
+const TableStyles = styled.div`
+  table {
+    background-color: ${({ theme }) => theme.colors.grayscale.light4};
+  }
+
+  .table > thead > tr > th {
+    border-bottom: ${({ theme }) => theme.gridUnit / 2}px solid
+      ${({ theme }) => theme.colors.grayscale.light2};
+    background: ${({ theme }) => theme.colors.grayscale.light4};
+  }
+`;
+
+const StyledTableStylesContainer = styled.div`
+  overflow: auto;
+`;
+function QuerySearch({ actions, displayLimit }: propTypes) {
+  const [databaseId, setDatabaseId] = useState<any>('');
+  const [userId, setUserId] = useState<string>('');
+  const [searchText, setSearchText] = useState<string>('');
+  const [from, setFrom] = useState<string>('28 days ago');
+  const [to, setTo] = useState<string>('now');
+  const [status, setStatus] = useState<string>('success');
+  const [queriesArray, setQueriesArray] = useState<any>([]);
+  const [queriesLoading, setQueriesLoading] = useState<boolean>(true);
+
+  const getTimeFromSelection = (selection: string) => {
+    switch (selection) {
+      case 'now':
+        return now();
+      case '1 hour ago':
+        return epochTimeXHoursAgo(1);
+      case '1 day ago':
+        return epochTimeXDaysAgo(1);
+      case '7 days ago':
+        return epochTimeXDaysAgo(7);
+      case '28 days ago':
+        return epochTimeXDaysAgo(28);
+      case '90 days ago':
+        return epochTimeXDaysAgo(90);
+      case '1 year ago':
+        return epochTimeXYearsAgo(1);
+      default:
+        return null;
+    }
+  };
+
+  const insertParams = (baseUrl: string, params: string[]) => {
+    const validParams = params.filter(function (p) {
+      return p !== '';
+    });
+    return `${baseUrl}?${validParams.join('&')}`;
+  };
+
+  const refreshQueries = async () => {
+    setQueriesLoading(true);
+    const params = [
+      userId && `user_id=${userId}`,
+      databaseId && `database_id=${databaseId}`,
+      searchText && `search_text=${searchText}`,
+      status && `status=${status}`,
+      from && `from=${getTimeFromSelection(from)}`,
+      to && `to=${getTimeFromSelection(to)}`,
+    ];
+
+    // make into async method
+    try {
+      const promise = await SupersetClient.get({
```

Comment:
you can name this response. It looks like it's returning an object rather than a promise.